destination/format/csv.go

```diff
@@ -94,6 +101,39 @@ func MakeCSVBytes(
 		}
 	}
 
+	// loop through records and de-dupe before converting to CSV
+	// this is done beforehand, so we can parallelize the CSV formatting
+	latestRecordMap := make(map[string]recordSummary, len(records))
+
+	for _, r := range records {
+		readAt := r.Metadata["opencdc.readAt"]
+		s, err := extract(r.Operation, r.Payload)
+		if err != nil {
+			return nil, err
+		}
+		key := fmt.Sprint(s[primaryKey]); 
+		l, ok := latestRecordMap[key]
+		if !ok {
+			l.latestRecord = &r
```

Comment:
@samirketema so `r` is a range scoped variable and is reused in the loop, if you want the ptr `&records[I]` is safer.